src/main/kotlin/io/emeraldpay/dshackle/upstream/AbstractHead.kt

```diff
@@ -154,9 +163,38 @@ abstract class AbstractHead @JvmOverloads constructor(
 
     override fun stop() {
         stopping = true
+        log.debug("Stop ${this.javaClass.simpleName} $upstreamId")
+        future?.let {
+            running[this.javaClass.simpleName]?.decrementAndGet()
+            it.cancel(true)
+        }
+        future = null
     }
 
     override fun start() {
         stopping = false
+        log.debug("Start ${this.javaClass.simpleName} $upstreamId")
+        if (future == null) {
+            future = executor.scheduleAtFixedRate(
+                {
+                    val delay = System.currentTimeMillis() - lastHeadUpdated
+                    delayed.set(delay > awaitHeadTimeoutMs)
+                    if (delayed.get()) {
+                        log.warn("No head updates $upstreamId for $delay ms @ ${this.javaClass.simpleName}")
+                    }
+                    System.gc()
```

Comment:
I think it can cause perfomance issues...
Maybe we can create some sort of queue for old heads?